docs/proposals/backup/backup.md

```diff
@@ -0,0 +1,735 @@
+	Destination *Destination `json:"destination,omitempty"`
+
+	// Policy are the rules defining how backups should be performed.
+	// +optional
+	Policy *BackupPolicy `json:"policy,omitempty"`
+}
+
+// Note: partly copied from https://github.com/vmware-tanzu/velero/pkg/apis/backup_types.go
+// BackupSpec defines the specification for a backup.
+type BackupPolicy struct {
+	// ResourceFilter specifies which resources should be included in the backup.
+	// It acts as a selective criterion to determine which resources are relevant for backup.
+	// If not set, the backup process will consider all resources. This filter helps in optimizing the backup process by excluding unnecessary data.
+	// +optional
+	ResourceFilter *ResourceFilter `json:"resourceFilter,omitempty"`
+
+	// TODO: consider SnapshotVolumes for backup
+
+	// TTL is a time.Duration-parseable string describing how long the Backup should be retained for.
+	// +optional
+	TTL metav1.Duration `json:"ttl,omitempty"`
+
+	// OrderedResources specifies the backup order of resources of specific Kind.
+	// The map key is the resource name and value is a list of object names separated by commas.
+	// Each resource name has format "namespace/objectname".  For cluster resources, simply use "objectname".
+	// For example, if you have a specific order for pods, such as "pod1, pod2, pod3" with all belonging to the "ns1" namespace,
+	// and a specific order for persistentvolumes, such as "pv4, pv8", you can use the orderedResources field in YAML format as shown below:
+	// ```yaml
+	// orderedResources:
+	//  pods: "ns1/pod1, ns1/pod2, ns1/pod3"
+	//  persistentvolumes: "pv4, pv8"
+	// ```
+	// +optional
+	// +nullable
+	OrderedResources map[string]string `json:"orderedResources,omitempty"`
+
+	// ItemOperationTimeout specifies the time used to wait for asynchronous BackupItemAction operations.
+	// The default value is 1 hour.
+	// +optional
+	ItemOperationTimeout metav1.Duration `json:"itemOperationTimeout,omitempty"`
+}
+
+type BackupStatus struct {
+	// Conditions represent the current state of the backup operation.
+	// +optional
+	Conditions capiv1.Conditions `json:"conditions,omitempty"`
+
+	// Phase represents the current phase of the backup operation.
+	// +optional
+	Phase string `json:"phase,omitempty"`
+
+	// Details provides a detailed status for each backup in each cluster.
+	// +optional
+	Details []*BackupDetails `json:"backupDetails,omitempty"`
+}
+
+type BackupDetails struct {
+	// ClusterName is the Name of the cluster where the backup is being performed.
+	// +optional
+	ClusterName string `json:"clusterName,omitempty"`
+
+	// ClusterKind is the kind of ClusterName recorded in Kurator.
+	// +optional
+	ClusterKind string `json:"clusterKind,omitempty"`
+
+	// BackupNameInCluster is the name of the backup being performed within this cluster.
+	// This BackupNameInCluster is unique in Storage.
+	// +optional
+	BackupNameInCluster string `json:"backupNameInCluster,omitempty"`
```

Comment:
wonder will this be automatically deleted when ttl expire

Reply:
If Velero sees that an existing backup resource is expired, it removes:

The backup resource
The backup file from cloud object storage
All PersistentVolume snapshots
All associated Restores
The effects of expiration are not applied immediately, they are applied when the gc-controller runs its reconciliation loop every hour by default
see https://velero.io/docs/v1.12/how-velero-works/#set-a-backup-to-expire